semver/format.ts

```diff
@@ -37,10 +26,6 @@ function formatNumber(value: number) {
  * @returns The string representation of a semantic version.
  */
 export function format(semver: SemVer): string {
-  if (isWildcardComparator(semver)) {
-    return "*";
-  }
-
```

Comment:
Why was this change made?

Reply:
This is also handling of non-standard semvers. If a semver is `{ major: NaN, minor: NaN, patch: NaN }`, then it returns `*`, which is not documented, nor tested. I don't think this is a reasonable behavior.

Reply:
SGTM